ratatui-widgets/src/table/cell.rs

```diff
@@ -113,6 +115,46 @@ impl<'a> Cell<'a> {
         self
     }
 
+    /// Set the `colspan` of this cell
+    ///
+    /// This is a fluent setter method which must be chained or used as it consumes self
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use ratatui::widgets::{Cell, Row};
+    ///
+    /// let rows = vec![
+    ///     Row::new(vec![
+    ///         Cell::new("Cell1Cell1Cell1").colspan(2),
+    ///         Cell::new("Cell2"),
+    ///     ]),
+    ///     Row::new(vec![
+    ///         Cell::new("Cell3Cell3Cell3"),
+    ///         Cell::new("Cell4Cell4Cell4").colspan(2),
+    ///         Cell::new("Cell5"),
+    ///     ]),
+    ///     Row::new(vec![
+    ///         Cell::new("Cell6").colspan(1),
+    ///         Cell::new("Cell7").colspan(1),
+    ///         Cell::new("Cell8"), // colspan(1) is the default
+    ///     ]),
+    /// ];
+    /// // "Cell1Cell1C Cell2",
+    /// // "Cell3 Cell4Cell4C",
+    /// // "Cell6 Cell7 Cell8",
+    /// ```
+    #[must_use = "method moves the value of self and returns the modified value"]
+    pub const fn colspan(mut self, colspan: u16) -> Self {
```

Comment:
Someone suggested `spans` could also be used as a name for this functionality. `colspan` seems abbreviated a bit too much. But it's not a biggie.